great_expectations/data_context/store/store.py

```diff
@@ -271,16 +279,15 @@ def build_store_from_config(
                 config_defaults=config_defaults,
             )
         except gx_exceptions.DataContextError as e:
-            new_store = None
             logger.critical(
                 f"Error {e} occurred while attempting to instantiate a store."
             )
-        if not new_store:
             class_name: str = store_config["class_name"]
-            module_name = store_config["module_name"]
+            module_name = store_config["module_name"]  # TODO: use .get() ???
             raise gx_exceptions.ClassInstantiationError(
                 module_name=module_name,
                 package_name=None,
                 class_name=class_name,
-            )
+            ) from e
+
```

Comment:
This should make is easier for user's to understand what went wrong.
Looking at user error messages in the support channel, the `logger.critical` message above doesn't always make it to the user.